_posts/2023-10-08-amazon_bedrock_part2.markdown

```diff
@@ -0,0 +1,130 @@
+---
+layout: postv2
+font: serif
+title: "Story telling with Amazon Bedrock"
+description: "Part2: Building your very own storytelling application with Foundation Models from Amazon Bedrock"
+date: 2023-10-08
+author: Nadia Reyhani
+tags: [Machine Learning, Amazon Web Services, ML, Generative AI, Amazon Bedrock, AWS]
+---
+
+## Initial Words
+
+Welcome to Part 2 of our introductory blog series! In Part 1, we delved into the basic concepts such as Foundation Models and discovered the amazing features and capabilities of Amazon Bedrock. Now, it's time for the fun part: building your very own storytelling application with Foundation Models from Amazon Bedrock.
+
+Just a heads-up, we're keeping things super simple in this app. We won't be focusing on the bells and whistles of user experience for now. I'll explain what I mean by that in a moment, but first, let's talk about what you'll achieve by the end of this post.
+
+By the time we're done here, you'll have deployed a serverless setup with two APIs. We'll be using Appsync and GraphQL to make requests to these APIs and generate stories. If you're wondering how these APIs work their magic with FM models to conjure up stories and illustrations, that's the enchantment of Amazon Bedrock we'll uncover together. So, let's get started on this storytelling adventure! 
+
+Now that you've got a glimpse of what we're building, let's take a moment to unravel the complexity hidden behind this seemingly simple feature. Imagine this: a user hops onto your web app and wants to create a story by giving it a topic prompt. From the user's standpoint, they expect the story to magically unfold, complete with illustrations, as quickly as they can think of it. They might want to edit and personalize the story, or even ensure that it suits the age group it's intended for. And what if they want to save and share their literary masterpiece with others?
+
+All these amazing features and optimizations are like extra layers of icing on the cake, but for our project, we're keeping things focused. So, while they're fascinating possibilities, we'll save them for another time!
+
+## Build an Storytelling app with me 
+
+Take a look at the cool diagram below! It shows you exactly how our app works at every step:
+
+- The user, starts by giving us an idea for a story or what the story should be about.
+
+- When user clicks on "Generate Story," the web app send a request to foundation model to create the story, and then it returns the generated story.
+
+- Now, here's where it gets interesting. user can either generate another story with the same topic or change the topic and get a new one. Plus, they can even add illustrations to the story! In this app, I've configured the FM model to generate an image for each paragraph. These images are stored in an S3 bucket, and the UI shows them to user once it gets back the S3 presigned URLs.
```

Comment:
` Now, here's where it gets interesting. user can either generate...`  -> ` Now, here's where it gets interesting. The user can either generate...`